baconclient/baconclient.go

```diff
@@ -308,26 +307,27 @@ func (b *BaconClient) CheckBakerRegistered() error {
 	return errors.New("Unknown error in determining baker status")
 }
 
-// Check if the balance of the baker is > 8001 Tez; Extra 1 Tez is for submitting reveal, if necessary
+// CheckDelegateBalance Check if the balance of the baker is > 8001 Tez; Extra 1 Tez is for submitting reveal, if necessary
 func (b *BaconClient) CheckDelegateBalance() error {
-
 	dbi := rpc.DelegateBalanceInput{
-		BlockID:  &rpc.BlockIDHead{},
+		BlockID:  new(rpc.BlockIDHead),
 		Delegate: b.Signer.BakerPkh,
 	}
 
 	resp, delegateBalance, err := b.Current.DelegateBalance(dbi)
 
-	log.WithFields(log.Fields{
-		"Request": resp.Request.URL, "Response": string(resp.Body()),
-	}).Trace("Fetching delegate balance")
+	if resp != nil {
```

Comment:
If we do this, then we can't see the URL that was fetched when it fails, nor can we see the response string

Reply:
this was just to avoid a NPE if resp == nil, it should preserve the previous behavior, no?